core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -104,6 +108,52 @@ class HistoryServer(
 
   initialize()
 
+  private[spark] class HistoryServlet(val appId: String) extends HttpServlet {
```

Comment:
can you rename this to something more specific?  eg., `EventLogDownloadServlet`?
also I think it can probably be `priviate[history]`